Dalamud/Interface/ImGuiNotification/INotification.cs

```diff
@@ -22,19 +22,42 @@ public interface INotification
+    /// </summary>
+    INotificationIcon? Icon { get; set; }
+
+    /// <summary>Gets or sets a texture wrap that will be used in place of <see cref="Icon"/> if set.</summary>
+    /// <remarks>
+    /// <para>A texture wrap set via this property will <b>NOT</b> be disposed when the notification is dismissed.
+    /// Use <see cref="IActiveNotification.SetIconTexture(IDalamudTextureWrap?)"/> or
     /// <see cref="IActiveNotification.SetIconTexture(Task{IDalamudTextureWrap?}?)"/> to use a texture, after calling
     /// <see cref="INotificationManager.AddNotification"/>. Call either of those functions with <c>null</c> to revert
-    /// the effective icon back to this property.</remarks>
-    INotificationIcon? Icon { get; set; }
+    /// the effective icon back to this property.</para>
+    /// <para>This property and <see cref="IconTextureTask"/> are bound together. If the task is not <c>null</c> but
+    /// <see cref="Task.IsCompletedSuccessfully"/> is <c>false</c> (because the task is still in progress or faulted,)
+    /// the property will return <c>null</c>. Setting this property will set <see cref="IconTextureTask"/> to a new
+    /// completed <see cref="Task{TResult}"/> with the new value as its result.</para>
+    /// </remarks>
+    public IDalamudTextureWrap? IconTexture { get; set; }
+
+    /// <summary>Gets or sets a task that results in a texture wrap that will be used in place of <see cref="Icon"/> if
+    /// available.</summary>
+    /// <remarks>
+    /// <para>A texture wrap set via this property will <b>NOT</b> be disposed when the notification is dismissed.
+    /// Use <see cref="IActiveNotification.SetIconTexture(IDalamudTextureWrap?)"/> or
+    /// <see cref="IActiveNotification.SetIconTexture(Task{IDalamudTextureWrap?}?)"/> to use a texture, after calling
+    /// <see cref="INotificationManager.AddNotification"/>. Call either of those functions with <c>null</c> to revert
+    /// the effective icon back to this property.</para>
+    /// <para>This property and <see cref="IconTexture"/> are bound together.</para>
+    /// </remarks>
+    Task<IDalamudTextureWrap?>? IconTextureTask { get; set; }
 
     /// <summary>Gets or sets the hard expiry.</summary>
     /// <remarks>
     /// Setting this value will override <see cref="InitialDuration"/> and <see cref="ExtensionDurationSinceLastInterest"/>, in that
     /// the notification will be dismissed when this expiry expires.<br />
     /// Set to <see cref="DateTime.MaxValue"/> to make only <see cref="InitialDuration"/> take effect.<br />
-    /// If neither <see cref="HardExpiry"/> nor <see cref="InitialDuration"/> is not MaxValue, then the notification
+    /// If both <see cref="HardExpiry"/> and <see cref="InitialDuration"/> are MaxValue, then the notification
     /// will not expire after a set time. It must be explicitly dismissed by the user of via calling
```

Comment:
while you're tidying docs:

```suggestion
    /// will not expire after a set time. It must be explicitly dismissed by the user or via calling
```

Reply:
:smh: